__tests__/app.test.js

```diff
@@ -137,4 +136,76 @@ describe("3.getReviewById", () => {
     }) 
 })
 
+describe("4.getCommentsById", () => {
+    describe("GET /api/reviews/:id/comments", () => {
+        test("returns an array of comments", () => {
+            return request(app)
+            .get("/api/reviews/2/comments")
+            .expect(200)
+            .then(res => {
+                let comments = res.body.comments;
+                expect(Array.isArray(comments)).toBe(true);
+                expect(comments).toHaveLength(3);
+            })
+        })
+        test("returns an empty array if id is valid but no comments are found", () => {
+            return request(app)
+            .get("/api/reviews/1/comments")
+            .expect(200)
+            .then(res => {
+                comments = res.body.comments;
+                expect(comments).toHaveLength(0);
+            })
+        })
+        test("returns a 404 error if input is not found", () => {
+            return request(app)
+            .get("/api/reviews/47/comments")
+            .expect(404)
+        })
+        test("returns a 400 error if input is not number type", () => {
+            return request(app)
+            .get("/api/reviews/dj/comments")
+            .expect(400)
+        })
+    })
+})
+
+describe("5.postCommentById", () => {
+    describe("POST /api/reviews/:id/comments", () => {
+        test("returns a comment object", () => {
+            return request(app)
+            .post("/api/reviews/1/comments")
+            .send({ username: "dav3rid", body: "Testing a body" })
+            .expect(201)
+            .then(res => {
+                let comment = res.body.comment
+                expect(typeof(comment)).toBe("object");
+            })
+        })
+        test("returns the posted comment", () => {
+            return request(app)
+            .post("/api/reviews/1/comments")
+            .send({ username: "dav3rid", body: "Testing a body" })
+            .expect(201)
+            .then(res => {
+                let comment = res.body.comment
+                expect(comment[0]).toMatchObject({
+                    comment_id: 7,
+                    body: 'Testing a body',
+                    review_id: 1,
+                    author: 'dav3rid',
+                    votes: 0,
+                    created_at: expect.any(String), //new Date()
+                });
+            })
+        })
+        test("returns a 400 Bad Request status if any query key/parameter is invalid or missing", () => {
+            return request(app)
+            .post("/api/reviews/1/comments")
+            .send({ username: "X" })
+            .expect(400)
+        })
```

Comment:
Include a test for an appropriate error in the case of an invalid `review_id`, both in the case that its in the wrong format and a correct id that is not in the database